Userland/Libraries/LibWeb/Bindings/Intrinsics.cpp

```diff
@@ -32,7 +33,11 @@ bool Intrinsics::is_exposed(StringView name) const
 Intrinsics& host_defined_intrinsics(JS::Realm& realm)
 {
     VERIFY(realm.host_defined());
-    return verify_cast<Bindings::PrincipalHostDefined>(*realm.host_defined()).intrinsics;
+
+    if (is<Bindings::PrincipalHostDefined>(*realm.host_defined()))
+        return static_cast<Bindings::PrincipalHostDefined const&>(*realm.host_defined()).intrinsics;
+
+    return *verify_cast<Bindings::SyntheticHostDefined>(*realm.host_defined()).synthetic_realm_settings.intrinsics;
```

Comment:
This truly is some fun dynamic casting.

Is it worth adding an intermeidary base class that has an `intrinsics()` getter for use by LibWeb?

Reply:
Indeed, typically something that I do avoid. I suppose if we do so I can also implement the other helpers like that as well, e.g: https://github.com/LadybirdBrowser/ladybird/blob/7ae7b9b695b498447a6f32b9019b132ec74e0c84/Userland/Libraries/LibWeb/HTML/Scripting/Environments.cpp#L139

Which does seem a bit more tasteful